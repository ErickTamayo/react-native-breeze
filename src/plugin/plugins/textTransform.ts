import { PluginFunction } from "./types";

export type PluginGroups = {
  transform: "uppercase" | "lowercase" | "capitalize" | "normal-case";
};

export const pattern = /^(?<transform>uppercase|lowercase|capitalize|normal-case)$/;

export const plugin: PluginFunction<PluginGroups> = ({ groups }) => {
  const { transform } = groups;
  return { textTransform: transform === "normal-case" ? "none" : transform };
};
