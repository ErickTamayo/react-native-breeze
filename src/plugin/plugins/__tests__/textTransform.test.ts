import {
  shouldEvaluateTheCorrectPatternTest,
  shouldMatchOutputSnapshot,
} from "../../utils/tests";
import { pattern, plugin } from "../textTransform";

describe("textTransform", () => {
  const shouldMatch = ["uppercase", "lowercase", "capitalize", "normal-case"];

  shouldEvaluateTheCorrectPatternTest(pattern, {
    shouldMatch,
    shouldNotMatch: ["unknown"],
  });

  shouldMatchOutputSnapshot(pattern, plugin, shouldMatch);
});
