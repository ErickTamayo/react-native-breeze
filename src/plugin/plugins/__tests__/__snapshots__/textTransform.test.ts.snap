// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`textTransform Should parse (capitalize) correctly 1`] = `
Object {
  "textTransform": "capitalize",
}
`;

exports[`textTransform Should parse (lowercase) correctly 1`] = `
Object {
  "textTransform": "lowercase",
}
`;

exports[`textTransform Should parse (normal-case) correctly 1`] = `
Object {
  "textTransform": "none",
}
`;

exports[`textTransform Should parse (uppercase) correctly 1`] = `
Object {
  "textTransform": "uppercase",
}
`;
